presentation/presentation.go

```diff
@@ -350,6 +343,17 @@ func (p *Presentation) AddSlideWithLayout(l SlideLayout) (Slide, error) {
 	return csld, nil
 }
 
+func removeChoicesWithPics(choices []*pml.CT_GroupShapeChoice) []*pml.CT_GroupShapeChoice {
+	for i := 0; i < len(choices); i++ {
+		chc := choices[i]
+		if len(chc.Pic) > 0 {
```

Comment:
Here's what bothers me. In the current implementation there is a `for` here instead of `if` and the `chc` is actually being rewritten each loop. This is what certainly causes the panic in this case, but just removing the loop changes the intended logic here. I am not sure that this should be done like this

Reply:
I see your point. There is an issue with the way I've refactored it. I have oversimplified it, well spotted.

Reply:
Shortening an array while you're iterating through it is a bit of a mindf**k. I improved the test to capture where my solution was going wrong (i.e. it was skipping choices) and simplified the `removeChoicesWithPics` function further.

Reply:
@mec07 yes, removing in the loop is awkward indeed. Not it looks nice, thank you!